### PR TITLE
batch for highland integration

### PR DESCRIPTION
**Pull request by buzzdecafe**

### CODE REVIEW

File: src/batch.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+module.exports = _curry2(_checkForMethod('batch', function batch(n, list) {
+  var idx = 0;
+  var length = list.length;
+  var acc = [];
+  if (n < 1) {
+    return length === 0 ? [] : [list];
+  }
+  while (idx < length) {
+    acc[acc.length] = _slice(list, idx, Math.min(idx + n, length));
```
Comment (CrossEye) [inline src/batch.js]: We really should move this check into `_slice`.  There's no reason that callers need to check the bounds before calling `_slice`.

**Comment by CrossEye** [comment]
:herb:  Better than either of our solutions in #774.

**Comment by davidchambers** [comment]
Of these names `splitEvery` is my clear favourite. It reads very well when partially applied: `R.splitEvery(2)`. Compare with `R.batch(2)` and `R.chunk(2)`, which are less descriptive.

File: src/batch.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * @example
+ *
+ *      R.batch(2, [1, 2, 3, 4, 5]); //=> [[1, 2], [3, 4], [5]]
+ */
+module.exports = _curry2(_checkForMethod('batch', function batch(n, list) {
+  var idx = 0;
+  var length = list.length;
+  var acc = [];
+  if (n < 1) {
+    return length === 0 ? [] : [list];
```
Comment (davidchambers) [inline src/batch.js]: This seems like two base cases. Can this be simplified?

Comment (buzzdecafe) [inline src/batch.js]: well, we have to decide what to do based on both the length of `list` and the value of `n`. so i don't know if there is a simplification here, but I am open to suggestions.

File: src/batch.js (hunk 1 above)
Comment (buzzdecafe) [inline src/batch.js]: i was very surprised to discover that i had to do that. so i agree, internal `_slice` should be smart enough to handle the bounds.

**Comment by buzzdecafe** [comment]
we good here?

**Comment by davidchambers** [comment]
I'm a strong :-1: on the name.

**Comment by CrossEye** [comment]
I'd rather take the opportunity while we're thinking about it to fix `_splice` and not put this call here.

Also I agree with @davidchambers on the name.  I'd be fine with either `chunk` or `splitEvery`.

File: src/batch.js (hunk 1 above)
Comment (davidchambers) [inline src/batch.js]: #1047

**Pull request closed without merging**